src/dialect/mssql/mssql-driver.ts

```diff
@@ -238,6 +242,19 @@ class MssqlConnection implements DatabaseConnection {
     return tediousIsolationLevel
   }
 
+  #cancelRequest<O>(request: MssqlRequest<O>): Promise<void> {
+    return new Promise((resolve) => {
+      request.request.once('requestCompleted', resolve)
+
+      const wasCanceled = this.#connection.cancel()
+
+      if (!wasCanceled) {
+        request.request.off('requestCompleted', resolve)
+        resolve(undefined)
+      }
```

Comment:
this is the case where there was nothing to cancel, the request completed already.